nipype/interfaces/spm/preprocess.py

```diff
@@ -22,14 +22,153 @@
+    thresh = traits.Float(0.5, usedefault=True,
+        field='subj.defaults.defaultsval.mflags.thresh',
+        desc='threshold used to create brain mask from segmented data');
+    reg = traits.Float(0.02, usedefault=True,
+        field='subj.defaults.defaultsval.mflags.reg',
+        desc='regularization value used in the segmentation');
+    # EPI unwarping for quality check
+    epi = File(copyfile=False, exists=True, mandatory=True,
+        field='subj.session.epi',
+        desc='EPI to unwarp');
+    matchvdm = traits.Bool(True, usedefault=True,
+        field='subj.matchvdm',
+        desc='match VDM to EPI');
+    sessname = Str('_run-', usedefault=True,
+        field='subj.sessname',
+        desc='VDM filename extension');
+    writeunwarped = traits.Bool(False, usedefault=True,
+        field='subj.writeunwarped',
+        desc='write unwarped EPI');
+    anat = File(copyfile=False, exists=True,
+        field='subj.anat',
+        desc='anatomical image for comparison');
+    matchanat = traits.Bool(True, usedefault=True,
+        field='subj.matchanat',
+        desc='match anatomical image to EPI');
+
+
+class FieldMapOutputSpec(TraitedSpec):
+    vdm = File(exists=True, desc='voxel difference map')
+
+
+class FieldMap(SPMCommand):
+    """Use the fieldmap toolbox from spm to calculate the voxel displacement map (VDM).
+
+    http://www.fil.ion.ucl.ac.uk/spm/doc/manual.pdf#page=173
+
+    To do
+    -----
+    Deal with real/imag magnitude images and with the two phase files case.
+
+    Examples
+    --------
+    >>> from nipype.interfaces.spm import FieldMap
+    >>> fm = FieldMap()
+    >>> fm.inputs.phase = 'phase.nii'
+    >>> fm.inputs.magnitude = 'magnitude.nii'
+    >>> fm.inputs.et = [5.19, 7.65]
+    >>> fm.inputs.blipdir = 1
+    >>> fm.inputs.tert = 15.6
+    >>> fm.inputs.epi = 'bold.nii'
+    >>> fm.run() # doctest: +SKIP
+
+    """
+
+    input_spec = FieldMapInputSpec
+    output_spec = FieldMapOutputSpec
+    _jobtype = 'tools'
+    _jobname = 'fieldmap'
+
+    def _format_arg(self, opt, spec, val):
+        """Convert input to appropriate format for spm
+        """
+        if opt in ['phase', 'magnitude', 'anat', 'epi']:
+            return scans_for_fname(filename_to_list(val))
+
+        return super(FieldMap, self)._format_arg(opt, spec, val)
+
+    def _parse_inputs(self):
+        """validate spm fieldmap options if set to None ignore
+        """
+        einputs = super(FieldMap, self)._parse_inputs()
+        jobtype = self.inputs.jobtype
+        return [{'%s' % (jobtype): einputs[0]}]
+
+    def _list_outputs(self):
+        outputs = self._outputs().get()
+        jobtype = self.inputs.jobtype
+        if jobtype == "calculatevdm":
+            outputs['vdm'] = []
+            for phase in filename_to_list(self.inputs.phase):
+                outputs['vdm'].append(fname_presuffix(phase, prefix='vdm5_sc'))
+            outputs['vdm'] = list_to_filename(outputs['vdm'])
```

Comment:
I'm a bit confused by this. `self.inputs.phase` is a single file, so this could just be:

```Python
if self.inputs.jobtype == 'calculatevdm':
    outputs['vdm'] = fname_presuffix(self.inputs.phase, prefix='vdm5_sc')
```

Or am I missing something?

Reply:
You're not, I just adapted another more complicated wrapper's piece of code without thinking about making it clearer. Thanks for noticing!